spec/unit/azurerm_server_create_spec.rb

```diff
@@ -1208,6 +1225,43 @@ class DummyClass < Azure::ResourceManagement::ARMInterface
       expect(extension["properties"]["settings"].has_key? 'extendedLogs').to be == false
     end
 
+    context "chef_service_interval option" do
+      context "is passed by the user" do
+        before do
+          @params[:chef_extension_public_param][:chef_service_interval] = "19"
+        end
+
+        it "sets the chef_service_interval parameter under extension config in the template" do
+          template = @service.create_deployment_template(@params)
+
+          extension = ""
+          template["resources"].each do |resource|
+            extension = resource if resource["type"] == "Microsoft.Compute/virtualMachines/extensions"
+          end
```

Comment:
Do we need to set extension to an empty string here? If it remains nil, wouldn't the expect statements fail either way?